src/standardSchema.ts

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copied from @standard-schema/spec to avoid having a dependency on it.
+ * https://github.com/standard-schema/standard-schema/blob/main/packages/spec/src/index.ts
+ */
+
+import type { AnySchema } from './types';
+import ValidationError from './ValidationError';
+
+export interface StandardSchema<Input = unknown, Output = Input> {
+  readonly '~standard': StandardSchemaProps<Input, Output>;
+}
+
+export interface StandardSchemaProps<Input = unknown, Output = Input> {
+  readonly version: 1;
+  readonly vendor: string;
+  readonly validate: (
+    value: unknown,
+  ) => StandardResult<Output> | Promise<StandardResult<Output>>;
+  readonly types?: StandardTypes<Input, Output> | undefined;
+}
+
+type StandardResult<Output> =
+  | StandardSuccessResult<Output>
+  | StandardFailureResult;
+
+interface StandardSuccessResult<Output> {
+  readonly value: Output;
+  readonly issues?: undefined;
+}
+
+interface StandardFailureResult {
+  readonly issues: ReadonlyArray<StandardIssue>;
+}
+
+interface StandardIssue {
+  readonly message: string;
+  readonly path?: ReadonlyArray<PropertyKey | StandardPathSegment> | undefined;
+}
+
+interface StandardPathSegment {
+  readonly key: PropertyKey;
+}
+
+interface StandardTypes<Input, Output> {
+  readonly input: Input;
+  readonly output: Output;
+}
+
+export function createStandardSchemaProps<TIn, Output>(
+  schema: AnySchema,
+): StandardSchemaProps<TIn, Output> {
+  /**
+   * Adapts the schema's validate method to the standard schema's validate method.
+   */
+  async function validate(value: unknown): Promise<StandardResult<Output>> {
+    try {
+      const result = await schema.validate(value, {
+        abortEarly: false,
+      });
+
+      return {
+        value: result as Output,
+      };
+    } catch (err) {
+      if (err instanceof ValidationError) {
+        return {
+          issues: issuesFromValidationError(err),
+        };
+      }
+
+      throw err;
+    }
+  }
+
+  return {
+    version: 1,
+    vendor: 'yup',
+    validate,
+  };
+}
+
+function createStandardPath(path: string | undefined): StandardIssue['path'] {
+  if (!path?.length) {
+    return undefined;
+  }
+
+  // Array to store the final path segments
+  const segments: string[] = [];
+  // Buffer for building the current segment
+  let currentSegment = '';
+  // Track if we're inside square brackets (array/property access)
+  let inBrackets = false;
+  // Track if we're inside quotes (for property names with special chars)
+  let inQuotes = false;
+
+  for (let i = 0; i < path.length; i++) {
```

Comment:
what is going on with path logic? There are already utilities for splitting and traversing yup's path's in utils, see the `reach` utility

Reply:
Here I'm not traversing the schema path which is what I think `reach` and `getIn` are for?

What I'm doing here is I'm converting error path from as single string value to an array of strings/symbols which is what the standard schema interface requires. 

For example these are the output paths from ValidationErrors and next to them is how they should look like in standard schema:

| Error Message Path | Array Format |
|-------------------|--------------|
| obj.foo | ['obj', 'foo'] |
| obj["not.obj.nested"] | ['obj', 'not.obj.nested'] |
| arr[0].foo | ['arr', '0', 'foo'] |
| arr[0]["not.array.nested"] | ['arr', '0', 'not.array.nested'] |
| ["not.a.field"] | ['not.a.field'] |

I scanned the codebase quickly, do we have utilities for this kind of thing?